### ISSUE TITLE
Logistics Pipes

### ISSUE THREAD

**Issue by LogicFalls** [GTNewHorizons/NewHorizons#3237]
... are better than Project Red: Transportation pipes, and should be added as an alternative if Minetweaker scripts comply.

**Comment by alphaNOVAircraft** [comment]
Is a pretty big mod, Project red is already in the pack, because the rest of the mod is useful (all the redstone stuff), but adding a whole new mod would make the pack bigger with just a improvement of a existing system. As far as i remember that mod was discussed often, but it never seemed that they will add it

**Comment by Dream-Master** [comment]
it was never added because minetweaker script not worked. But this was versions ago. Since much bc pipes are allready added I see nothing against it. Sure I will run a test first. But as always I need to hear more voices here.

**Comment by boubou19** [comment]
I've heard Logistics Pipes introduce a lot of lags, but it was long time ago so dunno if it's still true.

**Comment by alphaNOVAircraft** [comment]
Not sure if we need it, imo i do not see a reason for adding a mod that does nearly the same like project red but makes the pack even larger. I personally dont need it, but would use it if it would be in the game and aviable as early as the redpower ones are right now.

**Comment by Wolozo7** [comment]
I'd like to see it. The mod is certainly not essential due to Project Red, but after building a PR storage setup, it seems like a much more archaic version of LP.

**Comment by Dream-Master** [comment]
so far I rember we add project red because of the minetweaker isdue. LP was the first and PR the second choice.

**Comment by noobyaran9** [comment]
Lets add it and see how people think about it. Preferably add some conversion recipe between project red. If it really got accepted by majority, maybe we can consider removing project red: transportation.

**Comment by bartimaeusnek** [comment]
i dont think we need another pipe mod, since logistic pipes is 100% ripoff of project red. Im against it due to its laggyness and aswell bc of its dumb usage of reflexion codewise.

**Comment by Wolozo7** [comment]
What lag?

**Comment by boubou19** [comment]
Mostly because of the high amount going through the pipes, and having a lot of pipes.

**Comment by Ugachaga** [comment]
I've heard that LP is pretty powerful pipes. Can someone who understands the differences with Project Red pipes tell about them?

**Comment by alphaNOVAircraft** [comment]
@Ugachaga id does not actullay behaves so different, you just have some better ways especially in automatisation compared to project red. If you set it up right it can even be better then AE2, but due to the fact that it is actually transporting all the items in the pipes, and not just "generating" them where they need to go like ae2 does, it is way more laggy if build up for the same automations

**Comment by damug** [comment]
I am very fond of logistics bpipes and would much prefer to use it over project red. If I recall there are some uses of it that make it compelling to use in certain situations over ae2 as well, such as small stocking autocrafting setups for your main storage system. Basically it is amazing, and I think it is totally minetweakable these days.

**Comment by bengenie** [comment]
I do remeder playing on ftb infinity  so it should work but i prefrode ae2 over logistics pipe and ender oi logistics system

**Comment by Scottishprog** [comment]
Main issue with project red is the fact that a drawer system will show no more than a stack to a retriever. So you don't actually know how much you have in storage, which greatly reduces the utility.

**Comment by LogicFalls** [comment]
@Scottishprog
That's just one problem. Project Red has fewer capabilities, yet is somehow more complicated to understand than LP - not to mention poorly documented.

**Comment by Hakameda** [comment]
+1 Logistics pipes best mod.

**Comment by Hakameda** [comment]
It would also balance itself pretty well, Remove the gear recipes for pipes. Chipsets are MV gated. The most OP thing is the pipe that runs into an enderchest to req items anywhere but gated behind IV. You could drop it in and it would be fairly balanced with no gears

**Comment by Bryfer** [comment]
Iirc there is issues with these pipes and chunk borders?
Items get voided if the next chunk is not loaded, since the pipes gets full

I haven't used these pipes in a very long time tho so maybe it's fixed?

**Comment by Hakameda** [comment]
Not sure I've seen that recently, was using in BR a few months back and it
was fine

**Comment by AntidermisMC** [comment]
I'd be for adding LP as well, but moving PR pipes to LV and making LP a late MV/ early HV thing so they don't overlap, since IIRC PR pipes are already MV and then we get AE late HV.

**Comment by Hakameda** [comment]
Being gated at engravers is good. Takes some effort to get the MV
sensor/emitters

**Comment by Hakameda** [comment]
Don't think I've seen too many people use em anyway

**Comment by Dream-Master** [comment]
Yea i want to tier them like Project Red pipes but before Ae

**Comment by Scottishprog** [comment]
I think this would help overall... right now the PR pipes are too expensive for what they provide imho. Hopefully the LP will be priced so they actually get used before AE.

**Comment by Dream-Master** [comment]
If pr pipes to expensive why i not see a issue here? @Scottishprog

**Comment by Hakameda** [comment]
They are sort of junky, it's a shitty version of logistics pipes honestly

**Comment by Dream-Master** [comment]
So you suggest disable PR pipes and enable Logistics ?

**Comment by Scottishprog** [comment]
OR possibly make the PR pipe cheaper to make, with LP as a higher tier.  But just straight disabling PR pipes works as well.

**Comment by Dream-Master** [comment]
yea this need to be planed a bit. So no fast release here.

**Comment by Hakameda** [comment]
Enabling buildcraft pipes would be good too

**Comment by Dream-Master** [comment]
they are. Most of them.

**Comment by Vicus-N** [comment]
What else new can introduce Logistics Pipes? We already have GT Pipes, Ender IO Conduits, Project Red Pipes, Steve's Factory Manager and AE...

**Comment by noobyaran9** [comment]
>What else new can introduce Logistics Pipes? We already have GT Pipes, Ender IO Conduits, Project Red Pipes, Steve's Factory Manager and AE...

1) Replacement for the shitty PR pipes
2) Better stepping stone to AE

**Comment by boubou19** [comment]
don't disable PR pipes in the next release just make LP and PR cohabit for the next release so that players can test both and report issues if some are found

**Comment by Dream-Master** [comment]
ok

**Comment by Dream-Master** [comment]
@Vernam why ?

**Comment by Vernam** [comment]
All was already sad.. useless. So much mods we have in pack, so many opportunities we have. Anyway, there are those who want to make the game easier. it seems like the goal was to make the game complex, and not to simplify its cheap Applied system..

**Comment by Dream-Master** [comment]
Yes true too.

**Comment by Vernam** [comment]
So far I have not seen any decent argument for adding a new mod. it's not that difficult to reach ae2. I do not see the need for a new mod. And until I saw a worthy argument, some snot how difficult it is without ae

**Comment by Bryfer** [comment]
I have noticed that alot of player does not know alot about Projectred, and therefore we need to make some adjustments to the Projectred quests
I made a few suggestions already in #3277 Feel free to go there if you can think of anything else that should be added/changed

**Comment by damug** [comment]
I'd disagree, red power pipes are really annoying to use and quite janky. Logistics pipes can stand on their own quite well or be used in conjunction with an ae2 system to much success.  It's also not that difficult to bring autocrafting to mv if you tinker with the various components to tier it right which would ease a lot of the heart ache I've read over crafting machine components in the coins issue.

**Comment by Scottishprog** [comment]
Agreed. Bryfer's changes make it much less annoying/needlessly costly. However, RP pipes are bad compared to LP. Another way, I strongly prefer LP, and having RP has a cheap chest access system in LV/Late steam. The crafting volume are bad enough, that I think more automation sooner is necessary so the game as a whole does not drag too much. 

I understand that this is a "hard-mode" pack, however I think tedious activities should be minimized. (in preference to the "designing" portion.) That said, it's OK to encourage automation, but that automation needs to be reasonably available. RP currently is too expensive to set up for self-duplication at the LV tier, so it isn't an actual automation option. By the time it's (relative) costs drop, you are on the doorstep of AE, and might as well tough it out.

 I think that is the root of the issue for early tiers, the automation options aren't, which is bad for a pack designed around automation. I can't speak for higher tiers, since I have not reached them.

I think you can have RP in the LV tier, and a _reasonably_ priced LP for MV/early HV. Keep in mind when generating the recipes for these things, that they might have 100 pipes or so used, before HV.

tldr;  We need to encourage the use of automation early on, and it has to be practical, and LP is an opportunity to do just that.

**Comment by Vernam** [comment]
So what? most people are against LP in this discussion.

**Comment by noobyaran9** [comment]
>So what? most people are against LP in this discussion.

I suggest we let LP into the pack temporarily and see what everyone thinks and how well it performs. If it causes alot of issue, then we take it out. If it doesn't, i suggest that we take out project red : transportation as it is not that good compared to LP.

**Comment by mitchej123** [comment]
@Vernam Really? What do you base that on?  I'm for it but haven't said much here, mainly on discord.  What does the poll say?

**Comment by Vernam** [comment]
more than I am sure that the majority of voters do not even really understand what they are voting for. such people spit on the integrity of the modpack, they see "oh, new mod, let's" mindlessly voting. important arguments for the addition of mod, and not voted herd. So far, there were no worthy arguments for the mod, some whiners who reach hv is an achievement. It is not difficult to reach the applaid, there are alternatives to the applaid in the modpack. stop snotty nagging, because of how you make the modpack worse, light.

**Comment by Vernam** [comment]
I'm for rationality. and I do not see the rationality in adding this mod. there are alternatives. those who are complaining about complex alternatives - can you suggest a projectE then? or what will happen next? ask to put a legal X-ray? Modpack is meant as a heavy one, and this offer is a way to simplify it at times. The point is then to change the set of recipes for complicating the modpack, when one casual sentence violates the idea of ​​modpack and its essence

**Comment by mitchej123** [comment]
ProjectE and free EMC for everyone!

**Comment by noobyaran9** [comment]
The point of suggesting this mod is because LP is far superior than what project red can offers.

**Comment by noobyaran9** [comment]
>when one casual sentence violates the idea of ​​modpack and its essence

And which sentence violates the idea of modpack and its essence?

**Comment by Vernam** [comment]
copying the same mods on the functional, simplifying the game freeloaders and lazy people

**Comment by Vernam** [comment]
I will change my mind when I see the real need and meaning in the mod. for the time being I see whiners, who find it hard to even make a stove, and these people want to simplify the game to the level of project ozon 2.

**Comment by boubou19** [comment]
If it's too powerful, i trust Dream to nerf the recipes and/or disable very OP items. If you don't like LP then don't use it, it's always good to have multiple ways to accomplish the same thing. IMO Applied Energistics is overplayed but i'm not saying that we have to force players to use LP

**Comment by noobyaran9** [comment]
>copying the same mods on the functional, simplifying the game freeloaders and lazy people

Comment like this are the reason why discussion turns toxic. 

>copying the same mods on the functional

"Copying". LP may be similar to PR on the functionality, they're basically two different mod. And not to mention LP is far superior than what PR can do in term of functionality and enable the ability to use barrels and storage drawer within the LP system.

> simplifying the game freeloaders and lazy people

We're not going to let LP in without any changes or sufficient nerfing. How is this even simplifying the game freeloaders and lazy people? You will still need to construct the whole LP system.

**Comment by Vernam** [comment]
There are still no valid arguments where it would be clear that a mod is needed as air. at least. so far I see excuses. when I see the meaning in the mod - another matter. so far, someone is too lazy to look for alternatives and get to the applaid system, although this is not so difficult. LP works almost for nothing, which is already a wild simplification.

**Comment by Vernam** [comment]
give at least 3 arguments, that LP is necessary for the assembly. Is it difficult? except as a simplification of the game at the mv stage, this is not an argument. until there are at least three arguments - I see a suggestion how to make of gtnh - project ozone 2.

**Comment by boubou19** [comment]
then let lazy people play with AE you can't force them to play like you but keep in mind that there are also hardworkers in the GT:NH community and those ones will be happy to make contraptions with LP

**Comment by mitchej123** [comment]
License:

Project Red: https://github.com/MrTJP/ProjectRed/blob/master/LICENSE.md 
```Distribution of original or modified Mod
By default this Mod may only be distributed on its own by the Developers```

Logistics Pipes: https://github.com/RS485/LogisticsPipes/blob/dev/LICENSE.md
```Distribution of original or modified copy rights
The user is allowed to redistribute this mod partially, in totality, or included in a distribution.

When distributing binary files, the user must provide means to obtain its entire set of sources or modified sources at no costs.

All distributions of this mod must remain licensed under the MMPL.

All dependencies that this mod have on other mods or classes must be licensed under conditions comparable to this version of MMPL, with the exception of the Minecraft code and the mod loading framework (e.g. ModLoader, ModLoaderMP or Bukkit).

Modified version of binaries and sources, as well as files containing sections copied from this mod, should be distributed under the terms of the present license.```

**Comment by mitchej123** [comment]
>  except as a simplification of the game at the mv stage
Simplification?  You mean automation?  Automation is great!  More options for automation instead of pigeon holing everyone into the exact same setup is even better!

**Comment by boubou19** [comment]
IIRC logistics pipes need power to function, but PR don't so you need to tinker a way to power it right (last time i played with LP was before AE1 being released)

**Comment by Vernam** [comment]
all that is written about the pros - I think you can implement without the LP. just imagination. For the time being, I just saw a banal laziness in the basics. please, 3 worthy arguments for extreme necessity in mod. necessity, importance. because before this mod all played quietly, all suited.  Played for a long time without LP, so it is not so necessary

**Comment by mitchej123** [comment]
So basically you're dismissing everything we say;  You aren't the arbiter of whether an argument is worthy.

How about *you* provide 3 worthy arguments of why we shouldn't include it, and "we haven't had it forever and we played fine without it" is not a valid reason.

**Comment by boubou19** [comment]
everything doesn't have an extrem necessity, so stop rejecting every argument because you don't like LP. I'm entering UHV tier and always using AE as automation is borring and i already used PR a lot in HV so i would like to see diversity in this pack is it a bad thing?

**Comment by Vernam** [comment]
instead of a decent answer, do you only transfer the arrows to me? interestingly you are trying to argue the presence of mod. arguments before that were not enough, because they sounded like "I'm too lazy to automate everything, give me a toy easier". Yes, I'm not a judge, it's not up to me to decide whether the arguments are good. I'm just trying to get it read by the DreamMaster and saw in general the situation and the need. and is it worth it to add a mod. so I just put myself in his place and try to pull out from you extremely weighty arguments.

**Comment by boubou19** [comment]
The fact that there are no valid arguments according to you doesn't mean that the arguments aren't valid for others. If LP is added then just remove it on your side and that's it! Why are you concerned in "lazy" people pack's experience? Everyone plays like he wants.

**Comment by Vernam** [comment]
Here are my arguments against - LP lagging mod. He simplifies the game too much, which is not the idea of ​​modpack. new not needed mods only make modpack even harder, which reduces the number of people who can play it. too selfish. so it's more important how you prove that mod is extremely necessary than my arguments against. if everyone plays as they want - you can put the mod yourself, without requiring installation from everyone. it's selfish.

**Comment by boubou19** [comment]
Adding a mod isn't selfish because you can always avoid using it or even remove yourself but don't add a mod because only a few people disagree about it is selfish because not everyone knows adding a mod in a custom technic launcher or whatever launcher you use

**Comment by Ethryan** [comment]
Can we please stay on topic now?, 

This is a thread for suggesting a new mod, and getting a response from others in the community. 
Please take this discussion to discord if you want to continue it.

**Comment by boubou19** [comment]
lags issues are purely because too much items goes through pipes if you limit them it's fine

**Comment by Hakameda** [comment]
PR is a very shitty mock up of LP, If you find the value in PR. Logistics Pipes will do the same and about 10x better. It can be a replacement for AE if you didn't want to use it

On Mon, 9 Jul 2018 at 01:01, boubou19 <notifications@github.com<mailto:notifications@github.com>> wrote:

lags issues are purely because too much items goes through pipes if you limit them it's fine

**Comment by Vernam** [comment]
you do not even think about others and about the assembly. I'm sorry, but it's true. everything I thought about this mod - I wrote. Do not try to convince me, it's stupid. my opinion is my opinion. I wish the modpack to rise to a new level, rather than fall lower. I suggest closing the meaningless chatter, let's see what the Dream says

**Comment by boubou19** [comment]
my last words about this topic is that adding variety always add complexity because you have more materials to play with. But only making the pack harder for the only purpose of making it harder is pointless because i think the essence of the pack is that you have a constant progression of the difficulty so the more you progress, more the setups are interesting to make, and more easier the crafting tasks are. You can only use LP if it's you wish. It concerns only you.

**Comment by Hakameda** [comment]
There is amazing things you can make with logi pipes, In an automation based pack. That's valuable

On Mon, 9 Jul 2018 at 01:12, boubou19 <notifications@github.com<mailto:notifications@github.com>> wrote:

my last words about this topic is that adding variety always add complexity because you have more materials to play with. But only making the pack harder for the only purpose of making it harder is pointless because i think the essence of the pack is that you have a constant progression of the difficulty so the more you progress, more the setups are interesting to make, and more easier the crafting tasks are. You can only use LP if it's you wish. It concerns only you.

**Comment by CalterOK** [comment]
You often mention the extension of automation options to AE, and the fact that PR pipes are expensive, but for some reason no one remembered that automation as well as a quite convenient storage can be done with the help of auto and golems. You are obsessed with only the pipes and other options at the moment do not go see yourself.
I also did not like the PR pipes in view of their high prices and I found myself an alternative in magic. The repository used later went perfectly into the AE system without any problems. Therefore, I do not need any PR or LP pipes.

**Comment by Vernam** [comment]
We have translocators  and JABBA. With JABBA we can transfer items, yes :) we have ST to.  So much we have - so less we use..

**Comment by mitchej123** [comment]
>I play a lot with logistic pipes on infinity evolved and when my lp system grow to about 1k pipes and something about 150 autocraft recipes I start catch lag spikes every 20-40 seconds because lp recalculate path. 

TPS or FPS lag?

**Comment by Hakameda** [comment]
I've only ever had lag if I built a shitty system. Everything causes lag if
you set something up wrong or poorly

**Comment by Cinobi** [comment]
Don't listen to them, add this mod please! 10 tps with 10 players is too much, need to make it lower! =)

Last time i tryed this mod it was laggy and baggy as hell, not sure somthing changed.

**Comment by Bryfer** [comment]
If you wish to share your opinion about Logistics Pipes with us, please do it here:
The last git issue got really messy. Keep it clean and read the first message before posting.
https://github.com/GTNewHorizons/NewHorizons/issues/3286
